services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/ArionWingService.java

```diff
@@ -55,37 +54,27 @@ public void buildArionGatewayState (NetworkConfiguration networkConfiguration, U
                 builder.setMacAddress(arionWing.getMac());
                 gatewayStateBuilder.getConfigurationBuilder().addDestinations(builder);
             }
+            String id = UUID.randomUUID().toString();
+            gatewayStateBuilder.getConfigurationBuilder().setId(id);
             gatewayStateBuilder.getConfigurationBuilder().getArionInfoBuilder().setVni(vni);
-            gatewayStateBuilder.getConfigurationBuilder().getArionInfoBuilder().setSubnetId(subnet);
-            unicastGoalStateV2.getGoalStateBuilder().putGatewayStates(key, gatewayStateBuilder.build());
+            gatewayStateBuilder.getConfigurationBuilder().getArionInfoBuilder().setSubnetId(internalSubnetEntity.getId());
+            gatewayStateBuilder.getConfigurationBuilder().getArionInfoBuilder().setVpcId(internalSubnetEntity.getVpcId());
+            gatewayStateBuilder.getConfigurationBuilder().setGatewayType(Gateway.GatewayType.ARION);
+            gatewayStateBuilder.getConfigurationBuilder().getArionInfoBuilder().setPortInbandOperation(8300);
+            unicastGoalStateV2.getGoalStateBuilder().putGatewayStates(id, gatewayStateBuilder.build());
         }
     }
 
-    public String getArionGroup (int vni, String subnet) {
-        String key = String.valueOf(vni) + "-" + subnet;
-        Optional<SimpleNode> group = ring.locate(key);
+    public String getArionGroup (String subnetId) {
```

Comment:
Why remove the param of vni? Is this API happening inside a vpc, so you don't need the vni? 
I assume subnets could be the same across different vpcs.

Reply:
subnetId is UUID

Reply:
Make sense